qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -0,0 +1,179 @@
+        self.get_repo()
+        self.get_test_and_scratch_dirs_ready()
+        self.install_deps()
+
+        # NOTE: On teuthology machines it's necessary to run "make" as
+        # superuser since the repo is cloned somewhere in /tmp.
+        self.mount_a.client_remote.run(args=['sudo', 'make'],
+                                       cwd=self.repo_path, stdout=StringIO(),
+                                       stderr=StringIO())
+
+        self.mount_a.client_remote.run(args=['sudo', 'make', 'install'],
+                                       cwd=self.repo_path, omit_sudo=False,
+                                       stdout=StringIO(), stderr=StringIO())
+
+        self.create_reqd_users()
+        self.write_local_config()
+
+    def get_repo(self):
+        """
+        Clone xfstests_dev repository. If already present, update it.
+        """
+        from teuthology.orchestra import run
+
+        # TODO: make sure that repo is not cloned for every test. it should
+        # happen only once.
+        remoteurl = 'git://git.ceph.com/xfstests-dev.git'
+        self.repo_path = self.mount_a.client_remote.mkdtemp(suffix=
+                                                            'xfstests-dev')
+        self.mount_a.run_shell(['git', 'archive', '--remote=' + remoteurl,
+                                'HEAD', run.Raw('|'),
+                                'tar', '-C', self.repo_path, '-x', '-f', '-'])
+
+    def get_admin_key(self):
+        import configparser
+        from sys import version_info as sys_version_info
+
+        cp = configparser.ConfigParser()
+        if sys_version_info.major > 2:
+            cp.read_string(self.fs.mon_manager.raw_cluster_cmd(
+                'auth', 'get-or-create', 'client.admin'))
+        # TODO: remove this part when we stop supporting Python 2
+        elif sys_version_info.major <= 2:
+            cp.read_string(unicode(self.fs.mon_manager.raw_cluster_cmd(
+                'auth', 'get-or-create', 'client.admin')))
+
+        return cp['client.admin']['key']
+
+    def get_test_and_scratch_dirs_ready(self):
+        """ "test" and "scratch" directories are directories inside Ceph FS.
+            And, test and scratch mounts are path on the local FS where "test"
+            and "scratch" directories would be mounted. Look at xfstests-dev
+            local.config's template inside this file to get some context.
+        """
+        from os.path import join
+
+        self.test_dirname = 'test'
+        self.scratch_dirname = 'scratch'
+
+        # read var name as "test dir's mount path"
+        self.test_dirs_mount_path = self.mount_a.client_remote.mkdtemp(
+            suffix=self.test_dirname)
+        # read var name as "scratch dir's mount path"
+        self.scratch_dirs_mount_path = self.mount_a.client_remote.mkdtemp(
+            suffix=self.scratch_dirname)
+
+        self.mount_a.run_shell(['mkdir', self.test_dirname,
+                                self.scratch_dirname])
```

Comment:
Is it strictly necessary that `test_dirname` and `scratch_dirname` be in CephFS? Requiring three mount points (three clients) should be avoided. Alternatively, we can just use symlinks instead of two extra mounts?